plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy

```diff
@@ -257,9 +257,27 @@ class SchemaValidator extends PluginExtensionPoint {
                     continue;
                 }
                 def String type = '[' + get_param.type + ']'
+                def String enums_string = ""
+                if (get_param.enum != null) {
+                    def List enums = (List) get_param.enum
+                    def String chop_enums = ""
+                    for (val in enums) {
+                        def String add_val = chop_enums + ", " + val
+                        if (add_val.length() > dec_linewidth - 5) {
+                            chop_enums += ", ..."
+                        } else {
+                            if (chop_enums == "") {
+                                chop_enums += val
+                            } else {
+                                chop_enums += ", " + val
+                            }
+                        }
+                    }
```

Comment:
Feels a bit verbose - I wonder if we can do basically the same with a join? Something like this: (untested)
```suggestion
                    def String chop_enums = enums.join(", ")
                    if(chop_enums.length() > dec_linewidth){
                        chop_enums = chop_enums.substring(0, dec_linewidth-5) + ", ..."
                    }
```

Reply:
I tried to use `.join()` but didn't manage to chop the string as I wanted, I will try your suggestion, thanks!